### PR TITLE
[FLINK-6143] [clients] Fix unprotected access to this.flink in LocalE…

### PR DESCRIPTION
**Pull request by zhangminglei**
…xecutor#endSession.

Thanks for contributing to Apache Flink. Before you open your pull request, please take the following check list into consideration.
If your changes take all of the items into account, feel free to open your pull request. For more information and/or questions please refer to the [How To Contribute guide](http://flink.apache.org/how-to-contribute.html).
In addition to going through the list, please provide a meaningful description of your changes.

- [ ] General
  - The pull request references the related JIRA issue ("[FLINK-XXX] Jira title text")
  - The pull request addresses only one issue
  - Each commit in the PR has a meaningful commit message (including the JIRA id)

- [ ] Documentation
  - Documentation has been added for new functionality
  - Old documentation affected by the pull request has been updated
  - JavaDoc for public methods has been added

- [ ] Tests & Build
  - Functionality added by the pull request is covered by tests
  - `mvn clean verify` has been executed successfully locally or a Travis build has passed

### CODE REVIEW

File: flink-clients/src/main/java/org/apache/flink/client/LocalExecutor.java [hunk 1]
```diff
@@ -59,7 +59,7 @@
 	private final Object lock = new Object();
 
 	/** The mini cluster on which to execute the local programs */
-	private LocalFlinkMiniCluster flink;
+	private volatile LocalFlinkMiniCluster flink;
```
Comment (zentol) [inline flink-clients/src/main/java/org/apache/flink/client/LocalExecutor.java]: If we guard all accesses with locks this doesn't have to be volatile.

Comment (zhangminglei) [inline flink-clients/src/main/java/org/apache/flink/client/LocalExecutor.java]: Thanks for telling me so useful message. Appreciate it.

Comment (zhangminglei) [inline flink-clients/src/main/java/org/apache/flink/client/LocalExecutor.java]: @zentol

**Comment by zhangminglei** [comment]
@zentol I have updated the code. Thanks again.

**Comment by zentol** [comment]
merging.

**Comment by zhangminglei** [comment]
@zentol I am appreciate it.

**Pull request closed without merging**